witness/tests/Test/Dscp/Witness/Explorer/ExplorerSpec.hs

```diff
@@ -334,13 +334,24 @@ spec = describe "Explorer" $ do
                          then Nothing
                          else Just (skAddress sk, edCvMap)
 
-            let fairCv = FairCV . M.fromList $ catMaybes cvPairs
-            fairCvInvalid <- spoilFairCV fairCv
-
-            -- Do positive and negative case in one `it` to reduce test
-            -- running time
-            checkResGood <- lift $ checkFairCV fairCv
-            checkResBad <- lift $ checkFairCV fairCvInvalid
-            return $
-                fullyValid checkResGood &&
-                not (fullyValid checkResBad)
+            let cvPairs = catMaybes cvPairsMs
+            -- If we somehow generated an empty FairCV, just skip and
+            -- have a better luck next time.
+            if null cvPairs
+                then return $ property True
```

Comment:
Better use `pre`, it would make the test discarded rather than accepted (although this is minor here).